### DISCUSSION TITLE
cy.origin() not able to handle microsoft authentication

### DISCUSSION THREAD

**Discussion by rgangal** [cypress-io/cypress#21575] [General]
Hi, 
In my use case when I try to load an application is redirect to `https://login.microsoftonline.com/` for authentication. So far I was handling this issue by using selenium standalone script but `cy.origin` seems the solution for handling multi domains. 

But the issue I am facing while using cy.origin is that as soon as the site gets redirected to Microsoft authentication cypress loses control and cant interact with it. Here is code I am using: 

```Cypress.Commands.add('loginSession', (username, password) => {
  const args = { username, password };
  cy.session(args, () => {
    cy.visit({
      url: '',
      failOnStatusCode: false
    });

    cy.origin('https://login.microsoftonline.com/', { args }, ({ username, password }) => {
        cy.get('*[type="email"]').type(username);
        cy.get('*[type="submit"]').click();
        cy.get('*[value="Sign in"]').type(password);
        cy.get('*[value="Accept"]').click();
    });
  });
});
```

**Comment by muratkeremozcan** [comment]
This discussion https://github.com/cypress-io/cypress/discussions/21186#discussioncomment-2750821 and this issue https://github.com/cypress-io/cypress/issues/21307.
